### PR TITLE
Stop vendoring pluggy

### PR DESCRIPTION
**Pull request by tgoodlet**
If this is too much let me know guys ;)
Pretty sure I got everything but someone should double check.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def has_environment_marker_support():
 
 
 def main():
-    install_requires = ['py>=1.4.33', 'six>=1.10.0','setuptools']  # pluggy is vendored in _pytest.vendored_packages
+    install_requires = ['py>=1.4.33', 'six>=1.10.0','setuptools', 'pluggy==0.4.0']
```
Comment (nicoddemus) [inline setup.py]: Perhaps `'pluggy>=0.4.0,<0.5'`?

File: _pytest/_pluggy.py [hunk 2]
```diff
@@ -1,11 +1,6 @@
 """
-imports symbols from vendored "pluggy" if available, otherwise
-falls back to importing "pluggy" from the default namespace.
+Import symbols from ``pluggy``
```
Comment (nicoddemus) [inline _pytest/_pluggy.py]: I think we can get rid of this module altogether and just import `pluggy` docs normally no?

File: doc/en/writing_plugins.rst [hunk 3]
```diff
@@ -454,7 +454,7 @@ hook wrappers and passes the same arguments as to the regular hooks.
 
 At the yield point of the hook wrapper pytest will execute the next hook
 implementations and return their result to the yield point in the form of
-a :py:class:`CallOutcome <_pytest.vendored_packages.pluggy._CallOutcome>` instance which encapsulates a result or
+a :py:class:`CallOutcome <_pytest._pluggy._CallOutcome>` instance which encapsulates a result or
```
Comment (nicoddemus) [inline doc/en/writing_plugins.rst]: Could you please create the docs locally (`tox -e docs`) to ensure the changes links still work?

File: _pytest/_pluggy.py [hunk 4]
```diff
@@ -1,11 +1,6 @@
...
 """
 from __future__ import absolute_import, division, print_function
-try:
-    from _pytest.vendored_packages.pluggy import *  # noqa
-    from _pytest.vendored_packages.pluggy import __version__  # noqa
-except ImportError:
-    from pluggy import *  # noqa
-    from pluggy import __version__  # noqa
+from pluggy import *
+from pluggy import __version__
```
Comment (RonnyPfannschmidt) [inline _pytest/_pluggy.py]: we might want to consider just directly inlining the imports as a followup

Comment (nicoddemus) [reply _pytest/_pluggy.py]: I think we can do it in the same PR if @tgoodlet is willing

Comment (tgoodlet) [reply _pytest/_pluggy.py]: @nicoddemus @RonnyPfannschmidt yup no problem guys.
I'll take a look this morning.

Comment (nicoddemus) [reply _pytest/_pluggy.py]: Awesome, thanks!

File: doc/en/writing_plugins.rst (hunk 3 above)
Comment (tgoodlet) [reply doc/en/writing_plugins.rst]: @nicoddemus yep I'll test it out

Comment (tgoodlet) [reply doc/en/writing_plugins.rst]: seems to have worked 👍

**Comment by tgoodlet** [comment]
@RonnyPfannschmidt @nicoddemus alright made the changes.
If you guys want me to tighten up the history a bit let me know.

**Comment by RonnyPfannschmidt** [comment]
personally  i like the history intact, but please add a changelog fragment

**Comment by tgoodlet** [comment]
@nicoddemus @RonnyPfannschmidt I noticed in the `pip` docs they have

>Upgrading, removing, or adding a new vendored library gets a special mention using a news/<library>.vendor file. This is in addition to any features, bugfixes, or other kinds of news that pulling in this library may have. This uses the library name as the key so that updating the same library twice doesn’t produce two news file entries.

Are we doing that as well or do I just need to add a `.trivial` entry to the `changelog/` dir?

**Comment by nicoddemus** [comment]
IMO just the `trivial` entry; `pluggy` is such an internal part of pytest right now that most users are not even aware of it or its bug-fixes.

**Comment by tgoodlet** [comment]
@nicoddemus good enough?

**Comment by nicoddemus** [comment]
Yep, thanks!

**Comment by tgoodlet** [comment]
@nicoddemus I'm assuming those CI failures are expected?
I noticed some odd (hypothesis) failures when running `tox` locally.

**Comment by tgoodlet** [comment]
Thanks @The-Compiler :)

**Comment by tgoodlet** [comment]
Once you guys merge this I'll put up a new PR to bring in `pluggy` `0.5.0` which should give us speed improvements as we deprecate `__multicall__` support 😄

**Comment by RonnyPfannschmidt** [comment]
another item to consider, we could now cython-ize pluggy for cpython ^^
argument extraction can be sped up quite a bit

**Comment by tgoodlet** [comment]
@RonnyPfannschmidt good idea.
I'd love to take a stab at that.

**Pull request merged**